### PR TITLE
Renaming Instance to Partition.

### LINKED ISSUE #50: Change Instance(s) to cursor(s)
I'm finding several people are getting confused on what an instance is. It is a term that is already pretty well overloaded.

I think a better name might be cursor.

### PR DESCRIPTION
**Pull request by mpawliszyn**
Closes #50 

I'm also doing a migration in the database. If you think this is too far LMK.

### CODE REVIEW

File: client/src/main/proto/app/cash/backfila/client_service.proto [hunk 1]
```diff
@@ -30,16 +30,16 @@ message PrepareBackfillRequest {
   optional bool dry_run = 5;
 }
 message PrepareBackfillResponse {
-  message Instance {
-    optional string instance_name = 1;
+  message Partition {
+    optional string partition_name = 1;
```
Comment (shellderp) [inline client/src/main/proto/app/cash/backfila/client_service.proto:34]: it's all proto encoded so this should be safe

File: service/src/main/kotlin/app/cash/backfila/client/HttpClientServiceClientProvider.kt [hunk 2]
```diff
@@ -11,17 +14,14 @@ import okhttp3.Interceptor
...
-import javax.inject.Singleton
 
 @Singleton
 class HttpClientServiceClientProvider @Inject constructor(
-    @Named(HTTP) private val defaultHttpClientEndpointConfig: HttpClientEndpointConfig,
-    private val httpClientFactory: HttpClientFactory,
-    private val httpClientConfigUrlProvider: HttpClientConfigUrlProvider,
-    @HttpClientNetworkInterceptor private val networkInterceptors: List<Interceptor>,
-    private val moshi: Moshi
+  @Named(HTTP) private val defaultHttpClientEndpointConfig: HttpClientEndpointConfig,
```
Comment (shellderp) [inline service/src/main/kotlin/app/cash/backfila/client/HttpClientServiceClientProvider.kt:20]: im confused where 2 space hanging indent is coming from

File: service/src/main/resources/migrations/v12__backfila.sql [hunk 3]
```diff
@@ -0,0 +1,5 @@
+ALTER TABLE run_instances
```
Comment (shellderp) [inline service/src/main/resources/migrations/v12__backfila.sql:1]: looks good

File: client/src/main/proto/app/cash/backfila/client_service.proto (hunk 1 above)
Comment (mpawliszyn) [reply client/src/main/proto/app/cash/backfila/client_service.proto:34]: Yeah I believe so.

File: service/src/main/kotlin/app/cash/backfila/client/HttpClientServiceClientProvider.kt (hunk 2 above)
Comment (mpawliszyn) [reply service/src/main/kotlin/app/cash/backfila/client/HttpClientServiceClientProvider.kt:20]: Yeah me too. I'm not seeing it in intellij

File: service/src/main/resources/migrations/v12__backfila.sql (hunk 3 above)
Comment (mpawliszyn) [reply service/src/main/resources/migrations/v12__backfila.sql:1]: Yeah and tests work. Should be safe back to 5.6.

File: client/src/main/proto/app/cash/backfila/client_service.proto (hunk 1 above)
Comment (mpawliszyn) [reply client/src/main/proto/app/cash/backfila/client_service.proto:34]: I think the only potential minor problem is that any of the old javascript pages might have trouble post a deploy. A simple refresh will fix it.

File: service/src/main/resources/migrations/v12__backfila.sql (hunk 3 above)
Comment (shellderp) [reply service/src/main/resources/migrations/v12__backfila.sql:1]: mysql 5.6? who still uses that? :)

**Pull request merged**